libs/entitlement/src/main/java/org/elasticsearch/entitlement/initialization/EntitlementInitialization.java

```diff
@@ -241,16 +241,19 @@ private static PolicyManager createPolicyManager() {
             )
         );
 
-        Path trustStorePath = trustStorePath();
-        if (trustStorePath != null) {
+        // conditionally add FIPS entitlements if FIPS only functionality is enforced
+        if ("true".equals(System.getProperty("org.bouncycastle.fips.approved_only"))) {
+            // if custom trust store is set, grant read access to its location, otherwise use the default trust store
+            String trustStore = System.getProperty("javax.net.ssl.trustStore");
+            Path trustStorePath = trustStore != null ? Path.of(trustStore) : bootstrapArgs.libDir().resolve("security/jssecacerts");
```

Comment:
Isn't the lib dir the _Elasticsearch_ lib dir? Note the running jdk. For this doesn't it need to be relative to java home?

Reply:
absolutely right ... doh, this isn't covered by any tests, too easy to get it wrong

Reply:
fixed